server/src/main/java/io/druid/server/http/LookupCoordinatorResource.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.http;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.jaxrs.smile.SmileMediaTypes;
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+import com.metamx.common.IAE;
+import com.metamx.common.RE;
+import com.metamx.common.logger.Logger;
+import io.druid.audit.AuditInfo;
+import io.druid.audit.AuditManager;
+import io.druid.common.utils.ServletResourceUtils;
+import io.druid.guice.annotations.Json;
+import io.druid.guice.annotations.Smile;
+import io.druid.server.namespace.cache.LookupCoordinatorManager;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+/**
+ * Contains information about lookups exposed through the coordinator
+ */
+@Path("/druid/coordinator/v1/lookups")
+public class LookupCoordinatorResource
+{
+  private static final Logger LOG = new Logger(LookupCoordinatorResource.class);
+  private final LookupCoordinatorManager lookupCoordinatorManager;
+  private final @Smile ObjectMapper smileMapper;
+  private final @Json ObjectMapper jsonMapper;
+
+  @Inject
+  public LookupCoordinatorResource(
+      final LookupCoordinatorManager lookupCoordinatorManager,
+      final @Smile ObjectMapper smileMapper,
+      final @Json ObjectMapper jsonMapper
+  )
+  {
+    this.smileMapper = smileMapper;
+    this.jsonMapper = jsonMapper;
+    this.lookupCoordinatorManager = lookupCoordinatorManager;
+  }
+
+  @GET
+  @Produces({MediaType.APPLICATION_JSON, SmileMediaTypes.APPLICATION_JACKSON_SMILE})
+  public Response getTiers()
```

Comment:
This won't discover all the tiers. This function will return the view from the DB, this means it can be missing some tiers that you don't know about. It is better to list tiers via discovery.